### DISCUSSION TITLE
FileBrowser

### DISCUSSION THREAD

**Discussion by mariofix** [otto-torino/django-baton#252] [Ideas]
Hi, I've been using Baton for a while now and i really like it, I'm trying to make an idea and I need of a "File Browser", for now just a basic functionality to display any file in a specific directory, I've been trying django-filebrowser-no-grapelli but it doesn't play nice with Baton.
Has anyone used some sort of file browser with Baton? Is something like StaticFileExplorer in the roadmap?

Thanks!

**Comment by abidibo** [comment]
Hi @mariofix, personally I've never used a file browser in my apps. But I could definitely take a look at some third party packages and implement some sort of integration for them. Since I'm new to the world of django file browsers, which are the best packages available? Is django-filebrowser-no-grapelli the best choice?

**Comment by mariofix** [comment]
Cool!
 I'm also fairly new to django, so I wouldn't know if that one is the best , what I do know is that most of the plugins dealing with files in Django are for attaching files to a model, all of them are really cool. 
What i was looking for is this https://flask-admin.readthedocs.io/en/latest/api/mod_contrib_fileadmin/ as a way to browse through the static files from inside the admin panel.

I'm in the proccess of determining if I'm going to keep searching for a plugin that does this or build it myself (I'm inclining for the latter)

**Comment by abidibo** [comment]
Hi @mariofix ,

Baton supports django-filer (https://github.com/django-cms/django-filer) as file manager. As for file browser let me know if you develop your own package!

**Comment by mariofix** [comment]
Yeah, i think i'll write something closer to what i need, perhaps try to port FileAdmin from Flask Admin.

I'll let you know what i come up with

**Comment by abidibo** [comment]
I personally used cabinet (https://github.com/matthiask/django-cabinet) in one project, and after cloning the app with just a few adjustments I obtained a nice result. Maybe in the future I'll fork and republish it with all sort of integrations.
